docs/guides/latest/volumesnapshot/statefulset.md

```diff
@@ -0,0 +1,828 @@
+          dataSource:
+            kind: VolumeSnapshot
+            name: ${CLAIM_NAME}-${POD_ORDINAL}-1560231666
+           # name: ${CLAIM_NAME}-0-1560231666
+            apiGroup: snapshot.storage.k8s.io
+```
+Here,
+
+* `spec.target.volumeClaimTemplates`:
+  * `metadata.name` is the name of the restored `PVC` or prefix of the `VolumeSnapshot` name.
+  * `spec.dataSource`: `spec.dataSource` specifies the source of the data from where the newly created PVC will be intialized. It requires following fields to set:
+    * `apiGroup` is the group for resource being referenced. Now, kubernetes supports only `snapshot.storage.k8s.io`.
+    * `kind` is resource of the kind being referenced. Now, kubernetes supports only `VolumeSnapshot`.
+    * `name` is the `VolumeSnapshot` resource name. In `RestoreSession` crd, You can template the name by using the following convention:
+    ```
+     ${CLAIM_NAME}-${POD_ORDINAL}-<backup session creation timestamp in Unix epoch seconds>
+     ```  
+     or 
+     ```
+     ${CLAIM_NAME}-<backup session creation timestamp in Unix epoch seconds>
+     ```
+    `${CLAIM_NAME}` and `{POD_ORDINAL}` are resolved by the Stash operator and replaced by the `metadata.name` of the `volumeclaimtemplates` and pod ordinal respectively. You can set the snapshot name directly.
+
+Let's create the `RestoreSession` crd we have shown above.
+
+```console
+$ kubectl create -f ./docs/examples/volume-snapshot/statefulset/restoresession.yaml
+restoresession.stash.appscode.com/restore-pvc created
+````
+Once, you have created the `RestoreSession` crd, Stash will create a job to restore. We can watch the `RestoreSession` phase to check if the restore process is succeeded or not.
+
+Run the following command to watch RestoreSession phase,
+
+```console
+$ watch -n 1 kubectl get restore -n demo
+Every 1.0s: kubectl get restore -n demo                      suaas-appscode: Tue Jun 18 18:35:41 2019
+
+NAME          REPOSITORY-NAME   PHASE       AGE
+restore-pvc                     Running     10s
+restore-pvc                     Succeeded   1m
+```
+So, we can see from the output of the above command that the restore process succeeded.
+
+##### Verify Restored PVC:
+
+Once a restore process is complete, we will see that new PVCs with the name `source-pvc-stash-demo-0` , `source-pvc-stash-demo-1` and `source-pvc-stash-demo-2` has been created successfully.
+
+check that the status of the PVCs are bound,
+
+```console
+$  kubectl get pvc -n demo
+NAME                      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+source-pvc-stash-demo-0   Bound    pvc-7c773921-8c10-11e9-bd3e-42010a800011   6Gi        RWO            standard       10m
+source-pvc-stash-demo-1   Bound    pvc-7c7b32ad-8c10-11e9-bd3e-42010a800011   6Gi        RWO            standard       10m
+source-pvc-stash-demo-2   Bound    pvc-7c7d0bdf-8c10-11e9-bd3e-42010a800011   6Gi        RWO            standard       10m
+````
+
```

Comment:
Notice the `STATUS` field. `Bound` indicates that PVC has been initialized from the respective VolumeSnapshot.